pgmq-extension/sql/pgmq--1.3.3--1.3.4.sql

```diff
@@ -0,0 +1,66 @@
+-- Add function to convert archive queues to partitioned tables
+CREATE FUNCTION pgmq.convert_archive_partitioned(table_name TEXT,
+                                                 partition_interval TEXT DEFAULT '10000',
+                                                 retention_interval TEXT DEFAULT '100000',
+                                                 leading_partition INT DEFAULT 10)
+RETURNS void AS $$
+DECLARE
+a_table_name TEXT;
+a_table_name_old TEXT;
+qualified_a_table_name TEXT;
+qualified_a_table_name_old TEXT;
+BEGIN
+  PERFORM pgmq._ensure_pg_partman_installed();
+  SELECT FORMAT('%s', 'a_'|| table_name) INTO a_table_name;
+  SELECT FORMAT('%s', 'a_'|| table_name || '_old') INTO a_table_name_old;
+  SELECT FORMAT('%s.%s', 'pgmq','a_'|| table_name) INTO qualified_a_table_name;
+  SELECT FORMAT('%s.%s', 'pgmq','a_'|| table_name || '_old') INTO qualified_a_table_name_old;
+
+  PERFORM c.relkind
+    FROM pg_class c
+    JOIN pg_namespace n ON n.oid = c.relnamespace
+    WHERE c.relname = a_table_name
+    AND c.relkind = 'p';
+
+  IF FOUND THEN
+    RAISE NOTICE 'Table %s is already partitioned', a_table_name;
+    RETURN;
+  END IF;
+
+  PERFORM c.relkind
+    FROM pg_class c
+    JOIN pg_namespace n ON n.oid = c.relnamespace
+    WHERE c.relname = a_table_name
+    AND c.relkind = 'r';
+
+  IF NOT FOUND THEN
+    RAISE NOTICE 'Table %s doesnot exists', a_table_name;
+    RETURN;
+  END IF;
+
+  EXECUTE 'ALTER TABLE ' || qualified_a_table_name || ' RENAME TO ' || a_table_name_old;
+
+  EXECUTE 'CREATE TABLE '
+            || qualified_a_table_name
+            || '( msg_id BIGINT PRIMARY KEY,
+                  read_ct INT DEFAULT 0 NOT NULL,
+                  enqueued_at TIMESTAMP WITH TIME ZONE DEFAULT now() NOT NULL,
+                  archived_at TIMESTAMP WITH TIME ZONE DEFAULT now() NOT NULL,
+                  vt TIMESTAMP WITH TIME ZONE NOT NULL,
+                  message JSONB
+                ) PARTITION BY RANGE (msg_id)';
```

Comment:
Dose this table have the same format as the old one? If so, you can use `LIKE`, something like:

```sql
EXECUTE format(
    'CREATE TABLE pgmq.%I LIKE %s) PARTITION BY RANGE (msg_id)',
    'a_' || table_name, 'a_'|| table_name || '_old'
);
```